pkg/services/provisioning/provisioning_test.go

```diff
@@ -0,0 +1,59 @@
+package provisioning
+
+import (
+	"context"
+	"github.com/bmizerany/assert"
+	"github.com/grafana/grafana/pkg/services/provisioning/dashboards"
+	"github.com/grafana/grafana/pkg/setting"
+	"testing"
+	"time"
+)
+
+func TestProvisioningServiceImpl(t *testing.T) {
+	t.Run("Restart dashboard provisioning and stop service", func(t *testing.T) {
+		service, mock := setup()
+		ctx, cancel := context.WithCancel(context.Background())
+		var serviceRunning bool
+		var serviceError error
+		go func() {
+			serviceRunning = true
+			serviceError = service.Run(ctx)
+			serviceRunning = false
+		}()
+
+		assert.Equal(t, len(mock.Calls.PollChanges), 0, "PollChanges should not have been called")
+
+		service.ProvisionDashboards()
+		time.Sleep(time.Millisecond)
```

Comment:
Seems crude but did not find a better way to do this as `ProvisionDashboards` does the initial provisioning it does not wait until polling process is actually started.

Reply:
It's fine. Sometimes its the reasonable thing to do. When we do this we should rename the file `*_integration_test.go` and add an integration build flag `// +build integration` Ex https://github.com/grafana/grafana/blob/master/pkg/services/alerting/engine_integration_test.go#L1

This way we always run them in CI but dont have to block other tests in dev. When working on tests with the build flag you can always add `-tags=integration` in the CLI.

Reply:
Wouldn't that be confusing as this isn't an integration test? I mean this is all just testing the service internals not any kind of integration to other services. Or should the `// +build integration` be more about slower tests that could slow local testing?

Reply:
Sorry! I misread it.. I thought it was `time.Second` which is to slow IMO. for `time.Millisecond` it doesn't matter :)

Reply:
Yeah I hope thought it won't create some flaky tests. It seems to be enough locally. I assume there is nothing else to do and the waiting goroutine will just kick in right away without much delay.